### PR TITLE
molten-vk: update sha256

### PR DESCRIPTION
**Pull request by makigumo**
1.2.5 rerelease that fixed a regression

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by ZhongRuoyu** [comment]
Can you please check with upstream if the retagging was intentional?

File: Formula/m/molten-vk.rb [hunk 1]
```diff
@@ -2,10 +2,11 @@ class MoltenVk < Formula
   desc "Implementation of the Vulkan graphics and compute API on top of Metal"
   homepage "https://github.com/KhronosGroup/MoltenVK"
   license "Apache-2.0"
+  revision 1
```
Comment (chenrui333) [inline Formula/m/molten-vk.rb:5]: i dont think we need revision, just rebuild the bottle should be good enough.

**Comment by makigumo** [comment]
> Can you please check with upstream if the retagging was intentional?

All seems to be good. See answer below.

> Yes. The 1.2.5 release was updated.
> 
> The releases are intended to align with the corresponding Vulkan SDK release (in this case  1.3.261), are subject to change during SDK integration testing, and only finalized once the Vulkan SDK is released.
> 
> I apologize that this was not clarified anywhere. We have now marked the current release status as "RC2", will update it if any further changes go out, and will remove the RC designation once the SDK is released.
> 
> In future, we will continue to adopt this RC naming status until the corresponding SDK is released.
> 
> _Originally posted by @billhollings in https://github.com/KhronosGroup/MoltenVK/issues/1998#issuecomment-1683873097_
>

**Review by ZhongRuoyu — CHANGES_REQUESTED** [review]
> We have now marked the current release status as "RC2", will update it if any further changes go out, and will remove the RC designation once the SDK is released.
>
> In future, we will continue to adopt this RC naming status until the corresponding SDK is released.

IIUC, this suggests that this `1.2.5` release is still not final yet and retagging can happen again in the future. So I don't think we're ready to ship this yet.

I've added [a comment](https://github.com/KhronosGroup/MoltenVK/issues/1998#issuecomment-1684089007) in the upstream issue to discuss further.

(I was not "requesting changes" from you, @makigumo -- just wanted to prevent this from getting merged for now. Let's wait for upstream's response first.)

**Comment by ZhongRuoyu** [comment]
Let's hold this off for a while until the release is finalised. Luckily, [we probably won't have this issue again](https://github.com/KhronosGroup/MoltenVK/issues/1998#issuecomment-1684315802).

**Comment by makigumo** [comment]
It looks like the associated Vulkan SDK (1.3.261.0 ) has been released (https://vulkan.lunarg.com/sdk/home#mac).
The RC marking has been removed (https://github.com/KhronosGroup/MoltenVK/releases/tag/v1.2.5).
And the final sha256 matches the one in this PR.

So I would assume it's safe to proceed.

**Comment by billhollings** [comment]
> It looks like the associated Vulkan SDK (1.3.261.0 ) has been released (https://vulkan.lunarg.com/sdk/home#mac). The RC marking has been removed (https://github.com/KhronosGroup/MoltenVK/releases/tag/v1.2.5). And the final sha256 matches the one in this PR.
> 
> So I would assume it's safe to proceed.

Confirmed. That MoltenVK tag & release are now final.

**Review by ZhongRuoyu — APPROVED** [review]
Thanks for confirming, @makigumo and @billhollings.

**Comment by ZhongRuoyu** [comment]
I rebased the branch against `master` and squashed the commits into one. It should be good to go then.

**Review by ZhongRuoyu — APPROVED** [review]

**Pull request merged**